Events/UnhandledEvent.cs

```diff
@@ -12,8 +13,10 @@ public class UnhandledEvent : Event
 
         static UnhandledEvent()
         {
-        }
-
+             VARIABLES.Add("type", "The type of the unhandled event");
+        }   
+        
+        [JsonProperty("type")]
```

Comment:
This change is not needed... `UnhandledEvent` is used internally but isn't user facing except in the logs and the `edType` variable already defines the type of the event.

Reply:
Reversing this should fix the failing unit test.

Reply:
Change made